langchain/llms/openai.py

```diff
@@ -163,7 +163,7 @@ class BaseOpenAI(BaseLLM, BaseModel):
 
     def __new__(cls, **data: Any) -> Union[OpenAIChat, BaseOpenAI]:  # type: ignore
         """Initialize the OpenAI object."""
-        if data.get("model_name", "").startswith("gpt-3.5-turbo"):
+        if data.get("model_name", "").startswith(("gpt-3.5", "gpt-4")):
```

Comment:
Is there a reason to remove the `-turbo` here?

I think using `.startswith(("gpt-3.5-turbo", "gpt-4")):` would be a little more robust since we expect all _current_ gpt-3.5 family models to have the `-turbo` within the name